src/rgw/rgw_sal_motr.cc

```diff
@@ -3091,8 +3096,8 @@ int MotrObject::get_bucket_dir_ent(const DoutPrefixProvider *dpp, rgw_bucket_dir
 
   iter = bl.cbegin();
   ent.decode(iter);
-
-  obj_key = ent.key.name + '\a' + ent.key.instance;
+  key = get_rgw_obj_key(ent.key); // This done like this to fix compiler error
```

Comment:
Can we do it like this? can you check if it compiles?
```c++
key.set(ent.key);
```

Reply:
Good suggestion. I missed this one. Thanks!